storage/stores.go

```diff
@@ -225,3 +246,75 @@ func (ls *Stores) RangeLookup(key roachpb.RKey, _ *roachpb.RangeDescriptor, cons
 	}
 	return br.Responses[0].GetInner().(*roachpb.RangeLookupResponse).Ranges, nil
 }
+
+// ReadBootstrapInfo implements the gossip.Persistence interface. Read
+// attempts to read gossip bootstrap info from every known store and
+// finds the most recent from all stores to initialize the bootstrap
+// info argument. Returns nil if no stores contain gossip bootstrap
+// metadata or an error on any issues reading data from the stores.
+func (ls *Stores) ReadBootstrapInfo(bi *gossip.BootstrapInfo) error {
+	ls.biLatestTS = roachpb.ZeroTimestamp
```

Comment:
Also if this method returns early `ls.biLatestTS` will be left in a weird state. It should work with a local timestamp variable and only set `biLatestTS` once it has found a match (or not, since `updateAllBootstrapInfos` will currently just clobber it).